### PR TITLE
TMA-927: handle uppercase email inputs

### PR DESCRIPTION
**Pull request by kubamahnert**

### CODE REVIEW

File: spec/lcm/integration/user_provisioning_end_to_end_spec.rb [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+        m = muf_template.dup
+        m[:login] = u[:custom_login]
+        m[:project_id] = u[:project_id]
+        m
+      end.flatten
+
+      upload_user_filters_csv($mufs)
+    end
+
+    it 'setups the MUFs' do
```
Comment (yenkins) [inline spec/lcm/integration/user_provisioning_end_to_end_spec.rb]: Similar code found in user_provisioning_end_to_end_spec.rb:116, user_provisioning_end_to_end_spec.rb:190 (mass = 100)

File: spec/lcm/integration/user_provisioning_end_to_end_spec.rb [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+      @projects.each do |p|
+        filters = p.user_filters.to_a
+        expect(filters.length).to eq $user_amount
+        @users.each do |u|
+          expect(filters.find { |f| f.related.email == u })
+        end
+      end
+    end
+
+    it 'pushes them there' do
```
Comment (yenkins) [inline spec/lcm/integration/user_provisioning_end_to_end_spec.rb]: Similar code found in user_provisioning_end_to_end_spec.rb:116, user_provisioning_end_to_end_spec.rb:190 (mass = 100)

File: lib/gooddata/models/profile.rb [hunk 3]
```diff
@@ -214,7 +214,7 @@ def diff(user)
     #
     # @return [String] Email address
     def email
-      @json['accountSetting']['email'] || ''
+      @json['accountSetting']['email'].is_a?(String) ? @json['accountSetting']['email'].downcase : ''
```
Comment (yenkins) [inline lib/gooddata/models/profile.rb]: Calls '@json['accountsetting']' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/models/profile.rb]: Calls '@json['accountsetting']['email']' 2 times (DuplicateMethodCall)

File: lib/gooddata/models/user_filters/user_filter_builder.rb [hunk 4]
```diff
@@ -273,7 +273,7 @@ def self.maqlify_filters(filters, project_users, options = {})
         expression, errors = create_expression(f, labels_cache, lookups_cache, attrs_cache, options)
         safe_login = login.downcase
         profiles_uri = if options[:type] == :muf
-                         project_user = project_users.find { |u| u.login == login }
+                         project_user = project_users.find { |u| u.login == safe_login }
```
Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: Contains iterators nested 2 deep (NestedIterators)

Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: Has the variable name 'u' (UncommunicativeVariableName)

File: spec/lcm/integration/user_provisioning_end_to_end_spec.rb [hunk 5]
```diff
@@ -0,0 +1,215 @@
+require_relative 'support/constants'
+require_relative 'support/configuration_helper'
+require_relative 'support/connection_helper'
+require_relative 'support/lcm_helper'
+require_relative 'support/project_helper'
+require_relative 'support/s3_helper'
+
+def ensure_users_in_domain(logins )
+  logins.map do |l|
```
Comment (yenkins) [inline spec/lcm/integration/user_provisioning_end_to_end_spec.rb]: Has the variable name 'l' (UncommunicativeVariableName)

File: spec/lcm/integration/user_provisioning_end_to_end_spec.rb [hunk 6]
```diff
@@ -0,0 +1,215 @@
+require_relative 'support/constants'
+require_relative 'support/configuration_helper'
+require_relative 'support/connection_helper'
+require_relative 'support/lcm_helper'
+require_relative 'support/project_helper'
+require_relative 'support/s3_helper'
+
+def ensure_users_in_domain(logins )
```
Comment (yenkins) [inline spec/lcm/integration/user_provisioning_end_to_end_spec.rb]: Style/SpaceInsideParens: Space inside parentheses detected.

File: spec/lcm/integration/user_provisioning_end_to_end_spec.rb [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+  $muf_complexity = 1
+  $user_amount = 3
+
+  before(:all) do
+    @suffix = ConfigurationHelper.suffix
+    @rest_client = LcmConnectionHelper.production_server_connection
+    @domain = @rest_client.domain(LcmConnectionHelper.environment[:prod_organization])
+    $reuse_project = ENV['REUSE_PROJECT']
+    CACHE_DIR = 'spec/cache/simple_ldm_projects/'
+    if false
```
Comment (yenkins) [inline spec/lcm/integration/user_provisioning_end_to_end_spec.rb]: Lint/LiteralInCondition: Literal `false` appeared in a condition.

File: spec/lcm/integration/user_provisioning_end_to_end_spec.rb [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+  describe '1 - initial MUF setup' do
+    before do
+      ensure_users_in_domain @users
+
+      $user_data = @projects.map do |c|
+        @users.map do |u|
+          {
+            custom_login: u,
+            project_id: c.pid,
+            position: 'developer',
```
Comment (yenkins) [inline spec/lcm/integration/user_provisioning_end_to_end_spec.rb]: Style/TrailingCommaInLiteral: Avoid comma after the last item of a hash.

File: spec/lcm/integration/user_provisioning_end_to_end_spec.rb [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+      config_path = ConfigurationHelper.create_interpolated_tempfile(
+        @ub_template_path,
+        @test_context
+      )
+
+      $SCRIPT_PARAMS = JSON.parse(File.read(config_path))
+      GoodData::Bricks::Pipeline.users_brick_pipeline.call($SCRIPT_PARAMS)
+      @projects.each do |p|
+        users = p.users.to_a
+        expect(users.length).to eq ($user_amount + 1) # rubydev+admin is in the projects as well (as the admin)
```
Comment (yenkins) [inline spec/lcm/integration/user_provisioning_end_to_end_spec.rb]: Lint/ParenthesesAsGroupedExpression: `(...)` interpreted as grouped expression.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/398/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1245/ : FAILURE in 2m 56s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1515/ : SUCCESS in 2m 49s

File: spec/lcm/integration/user_provisioning_end_to_end_spec.rb [hunk 10]
```diff
@@ -0,0 +1,207 @@
+require_relative 'support/constants'
+require_relative 'support/configuration_helper'
+require_relative 'support/connection_helper'
+require_relative 'support/lcm_helper'
+require_relative 'support/project_helper'
+require_relative 'support/s3_helper'
+
+def ensure_users_in_domain(logins )
```
Comment (Hahihula) [inline spec/lcm/integration/user_provisioning_end_to_end_spec.rb]: the domain method add_user already contains this logic... maybe try to use it

File: spec/lcm/integration/user_provisioning_end_to_end_spec.rb [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+        m[label.title] = value
+        m
+      end.flatten
+
+      Support::S3Helper.upload_file(ConfigurationHelper.csv_from_hashes(@mufs), Support::S3Helper::USER_FILTERS_KEY)
+    end
+
+    it 'setups the MUFs' do
+      BrickRunner.user_filters_brick context: @test_context, template_path: @ufb_template_path
+      @projects.each do |p|
```
Comment (yenkins) [inline spec/lcm/integration/user_provisioning_end_to_end_spec.rb]: Identical code found in user_provisioning_end_to_end_spec.rb:101, user_provisioning_end_to_end_spec.rb:148 (mass = 124)

File: spec/lcm/integration/user_provisioning_end_to_end_spec.rb [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+        mdup
+      end
+
+      Support::S3Helper.upload_file(ConfigurationHelper.csv_from_hashes(mufs), Support::S3Helper::USER_FILTERS_KEY)
+    end
+  end
+
+  describe '3 - subsequent MUF setup' do
+    it 'checks if they appeared after adding users to project' do
+      @projects.each do |p|
```
Comment (yenkins) [inline spec/lcm/integration/user_provisioning_end_to_end_spec.rb]: Identical code found in user_provisioning_end_to_end_spec.rb:101, user_provisioning_end_to_end_spec.rb:148 (mass = 124)

File: spec/lcm/integration/brick_runner.rb [hunk 13]
```diff
@@ -0,0 +1,25 @@
+class BrickRunner
```
Comment (yenkins) [inline spec/lcm/integration/brick_runner.rb]: Has no descriptive comment (IrresponsibleModule)

File: spec/lcm/integration/user_provisioning_end_to_end_spec.rb [hunk 14]
```diff
@@ -0,0 +1,172 @@
...
+require_relative 'support/project_helper'
+require_relative 'support/s3_helper'
+require_relative 'brick_runner'
+
+def uniq_label_with_value(project)
+  blacklists = $generated_uniq_labels_with_value || {}
+  blacklist = blacklists[project.pid] || []
+
+  atts = project.attributes.to_a
+  attribute = atts.shift while !attribute || attribute.date_attribute? || !attribute.primary_label || (blacklist.include? attribute)
```
Comment (yenkins) [inline spec/lcm/integration/user_provisioning_end_to_end_spec.rb]: Calls 'attribute.primary_label' 2 times (DuplicateMethodCall)

File: spec/lcm/integration/user_provisioning_end_to_end_spec.rb [hunk 15]
```diff
@@ -0,0 +1,172 @@
...
+require_relative 'support/configuration_helper'
+require_relative 'support/connection_helper'
+require_relative 'support/lcm_helper'
+require_relative 'support/project_helper'
+require_relative 'support/s3_helper'
+require_relative 'brick_runner'
+
+def uniq_label_with_value(project)
+  blacklists = $generated_uniq_labels_with_value || {}
+  blacklist = blacklists[project.pid] || []
```
Comment (yenkins) [inline spec/lcm/integration/user_provisioning_end_to_end_spec.rb]: Calls 'project.pid' 2 times (DuplicateMethodCall)

File: spec/lcm/integration/user_provisioning_end_to_end_spec.rb [hunk 16]
```diff
@@ -0,0 +1,172 @@
+require_relative 'support/constants'
+require_relative 'support/configuration_helper'
+require_relative 'support/connection_helper'
+require_relative 'support/lcm_helper'
+require_relative 'support/project_helper'
+require_relative 'support/s3_helper'
+require_relative 'brick_runner'
+
+def uniq_label_with_value(project)
```
Comment (yenkins) [inline spec/lcm/integration/user_provisioning_end_to_end_spec.rb]: Has approx 10 statements (TooManyStatements)

Comment (yenkins) [inline spec/lcm/integration/user_provisioning_end_to_end_spec.rb]: Doesn't depend on instance state (maybe move it to another class?) (UtilityFunction)

File: spec/lcm/integration/user_provisioning_end_to_end_spec.rb [hunk 17]
```diff
@@ -0,0 +1,172 @@
...
+  blacklists = $generated_uniq_labels_with_value || {}
+  blacklist = blacklists[project.pid] || []
+
+  atts = project.attributes.to_a
+  attribute = atts.shift while !attribute || attribute.date_attribute? || !attribute.primary_label || (blacklist.include? attribute)
+  label = attribute.primary_label
+  value = label.values.find { |v| !v[:value].empty? }
```
Comment (yenkins) [inline spec/lcm/integration/user_provisioning_end_to_end_spec.rb]: Has the variable name 'v' (UncommunicativeVariableName)

File: spec/lcm/integration/brick_runner.rb [hunk 18]
```diff
@@ -0,0 +1,25 @@
...
+
+      config_path = ConfigurationHelper.create_interpolated_tempfile(
+        template_path,
+        test_context
+      )
+      JSON.parse(File.read(config_path))
+    end
+  end
+end
+
```
Comment (yenkins) [inline spec/lcm/integration/brick_runner.rb]: Style/TrailingBlankLines: 1 trailing blank lines detected.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/409/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1254/ : FAILURE in 1m 17s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1524/ : SUCCESS in 1m 02s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/410/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1255/ : FAILURE in 3m 50s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1525/ : SUCCESS in 2m 59s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/413/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1258/ : SUCCESS in 1m 06s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1528/ : SUCCESS in 1m 00s

**Comment by kubamahnert** [comment]
ok to test

**Comment by kubamahnert** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/414/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1259/ : SUCCESS in 1m 16s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1529/ : SUCCESS in 1m 07s

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/415/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1476/ : SUCCESS in 14m 25s
- https://checklist.intgdc.com/job/lcm/job/gooddata-ruby-lcm-integration-tests-zuul/48/ : FAILURE in 1m 23s

**Comment by Hahihula** [comment]
extended test

**Comment by Hahihula** [comment]
ok to merge

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/416/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1477/ : SUCCESS in 13m 39s
- https://checklist.intgdc.com/job/lcm/job/gooddata-ruby-lcm-integration-tests-zuul/49/ : FAILURE in 1m 08s

**Comment by Hahihula** [comment]
extended test staging

**Comment by kubamahnert** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/418/ : Tests on opensource repositories should be triggered by users in `gooddata` organization in 7s
- gooddata-ruby-integration-tests-zuul : SKIPPED
- gooddata-ruby-lcm-integration-tests-zuul : SKIPPED

**Comment by Hahihula** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/424/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1478/ : SUCCESS in 15m 00s
- https://checklist.intgdc.com/job/lcm/job/gooddata-ruby-lcm-integration-tests-zuul/50/ : SUCCESS in 43m 52s

File: lib/gooddata/models/profile.rb (hunk 3 above)
Comment (yenkins) [inline lib/gooddata/models/profile.rb]: info: Calls '@json['accountsetting']' 2 times (DuplicateMethodCall) (reek)

Comment (yenkins) [inline lib/gooddata/models/profile.rb]: info: Calls '@json['accountsetting']['email']' 2 times (DuplicateMethodCall) (reek)

File: lib/gooddata/models/user_filters/user_filter_builder.rb (hunk 4 above)
Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: info: Contains iterators nested 2 deep (NestedIterators) (reek)

Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: info: Has the variable name 'u' (UncommunicativeVariableName) (reek)

File: spec/lcm/integration/brick_runner.rb [hunk 19]
```diff
@@ -0,0 +1,24 @@
+class BrickRunner
```
Comment (yenkins) [inline spec/lcm/integration/brick_runner.rb]: info: Has no descriptive comment (IrresponsibleModule) (reek)

File: spec/lcm/integration/user_provisioning_end_to_end_spec.rb [hunk 20]
```diff
@@ -0,0 +1,158 @@
...
+require_relative 'support/project_helper'
+require_relative 'support/s3_helper'
+require_relative 'brick_runner'
+
+def uniq_label_with_value(project)
+  blacklists = $generated_uniq_labels_with_value || {}
+  blacklist = blacklists[project.pid] || []
+
+  atts = project.attributes.to_a
+  attribute = atts.shift while !attribute || attribute.date_attribute? || !attribute.primary_label || (blacklist.include? attribute)
```
Comment (yenkins) [inline spec/lcm/integration/user_provisioning_end_to_end_spec.rb]: info: Calls 'attribute.primary_label' 2 times (DuplicateMethodCall) (reek)

File: spec/lcm/integration/user_provisioning_end_to_end_spec.rb [hunk 21]
```diff
@@ -0,0 +1,158 @@
...
+require_relative 'support/configuration_helper'
+require_relative 'support/connection_helper'
+require_relative 'support/lcm_helper'
+require_relative 'support/project_helper'
+require_relative 'support/s3_helper'
+require_relative 'brick_runner'
+
+def uniq_label_with_value(project)
+  blacklists = $generated_uniq_labels_with_value || {}
+  blacklist = blacklists[project.pid] || []
```
Comment (yenkins) [inline spec/lcm/integration/user_provisioning_end_to_end_spec.rb]: info: Calls 'project.pid' 2 times (DuplicateMethodCall) (reek)

File: spec/lcm/integration/user_provisioning_end_to_end_spec.rb [hunk 22]
```diff
@@ -0,0 +1,158 @@
+require_relative 'support/constants'
+require_relative 'support/configuration_helper'
+require_relative 'support/connection_helper'
+require_relative 'support/lcm_helper'
+require_relative 'support/project_helper'
+require_relative 'support/s3_helper'
+require_relative 'brick_runner'
+
+def uniq_label_with_value(project)
```
Comment (yenkins) [inline spec/lcm/integration/user_provisioning_end_to_end_spec.rb]: info: Has approx 10 statements (TooManyStatements) (reek)

Comment (yenkins) [inline spec/lcm/integration/user_provisioning_end_to_end_spec.rb]: info: Doesn't depend on instance state (maybe move it to another class?) (UtilityFunction) (reek)

File: spec/lcm/integration/user_provisioning_end_to_end_spec.rb [hunk 23]
```diff
@@ -0,0 +1,158 @@
...
+  blacklists = $generated_uniq_labels_with_value || {}
+  blacklist = blacklists[project.pid] || []
+
+  atts = project.attributes.to_a
+  attribute = atts.shift while !attribute || attribute.date_attribute? || !attribute.primary_label || (blacklist.include? attribute)
+  label = attribute.primary_label
+  value = label.values.find { |v| !v[:value].empty? }
```
Comment (yenkins) [inline spec/lcm/integration/user_provisioning_end_to_end_spec.rb]: info: Has the variable name 'v' (UncommunicativeVariableName) (reek)

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/123/ : SUCCESS in 1m 02s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1356/ : SUCCESS in 2m 36s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1626/ : SUCCESS in 2m 37s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1498/ : SUCCESS in 15m 08s
- https://checklist.intgdc.com/job/lcm/job/gooddata-ruby-lcm-integration-tests-zuul/69/ : SUCCESS in 39m 52s

**Pull request merged**